services/_internal/maven-EGI/workflowEGI/src/main/java/it/infn/ct/DescribeVM.java

```diff
@@ -0,0 +1,329 @@
+                                if (bridge != null) System.out.println("org.opennebula.network.bridge = " + bridge);
+                        }
+                }
+
+		if (properties.getProperty("RESOURCE").equals("os_tpl"))
+		{
+			System.out.println("[ OS_TPL DESCRIPTION ]");
+			System.out.println("[[ " + properties.getProperty("OCCI_RESOURCE_ID") + " ]]");
+			String TERM = properties.getProperty("OCCI_RESOURCE_ID")
+				      .substring(properties.getProperty("OCCI_RESOURCE_ID").lastIndexOf("#") + 1);
+			List<Mixin> mixins = model.findRelatedMixins("os_tpl");
+		        for (Mixin mixin : mixins) {
+				if ( ((mixin.getTerm()).contains(TERM)) ) {
+					System.out.println("title: \t\t" + mixin.getTitle());
+			                System.out.println("term: \t\t" + mixin.getTerm());
+			                System.out.println("location: \t" + mixin.getLocation());
+				}
+			}
+		} 
+		
+		if (properties.getProperty("RESOURCE").equals("resource_tpl"))
+		{
+			// Getting the description of all the available template(s)
+			System.out.println("[ RESOURCE_TPL DESCRIPTION ]");
+			List<Mixin> mixins = model.findRelatedMixins(properties.getProperty("RESOURCE"));
+
+			if (!mixins.isEmpty())
+				for (Mixin mixin : mixins) {
+					if (mixin.getTerm().equals(properties.getProperty("OCCI_RESOURCE_ID"))) {
+						System.out.println("[[ " + mixin.getLocation() + " ]]");
+						System.out.println("title: \t\t" + mixin.getTitle());
+						System.out.println("term: \t\t" + mixin.getTerm());
+						String locations = (mixin.getLocation()).toString();
+						String segments[] = locations.split("/");
+						System.out.println("location: \t" + "/" + segments[segments.length - 1] + "/");
+					}
+				}
+		}
+
+	} catch (CommunicationException | RenderingException | AmbiguousIdentifierException ex) {
+		throw new RuntimeException(ex);}
+	}
+
+	public static String[] describe (String vmID)
+	{
+
+	// [ Setting preferences here! ]
+	String AUTH = "x509"; 
+	String OCCI_ENDPOINT_HOST = "https://carach5.ics.muni.cz:11443"; 
+
+	String TRUSTED_CERT_REPOSITORY_PATH = "/etc/grid-security/certificates";
+	String PROXY_PATH = "/tmp/x509up_u5040"; // <= Change here!
+
+	Boolean verbose = true;
+
+	String ACTION = "describe";	
+
+	// CESNET-MetaCloud
+	// [ *Describing* available resources (e.g. os_tpl, resource_tpl, compute, storage and network) ]
+	//List<String> RESOURCE = Arrays.asList("os_tpl",
+	//"http://occi.carach5.ics.muni.cz/occi/infrastructure/os_tpl#uuid_training_centos_6_fedcloud_warg_123"); // <= Change here!
+	
+	//List<String> RESOURCE = Arrays.asList("resource_tpl", "medium");
+
+	//List<String> RESOURCE = Arrays.asList("storage",
+	//"https://carach5.ics.muni.cz:11443/storage/3155"); // <= Change here!
+
+	//List<String> RESOURCE = Arrays.asList("network",
+	//"https://carach5.ics.muni.cz:11443/network/6"); // <= Change here!
+
+	List<String> RESOURCE = Arrays.asList("compute",
+	vmID); // <= Change here!
+	
+
+	// INFN-CATANIA-STACK
```

Comment:
You are exposing a lot of hard-coded URLs and port numbers. Make sure it is not going to cause any security problems. Having hard coded URLs in not a good idea (unless there is no other choice). A single common config file may be a solution